query/src/main/java/io/keen/client/java/RequestUrlBuilder.java

```diff
@@ -35,27 +38,60 @@
 
     /**
      * Get a formatted URL for an analysis request.
-     * 
-     * @param projectId The project id
-     * @param analysisPath The analysis url sub-path
      *
+     * @param projectId    The project id
+     * @param analysisPath The analysis url sub-path
      * @return The complete URL.
      * @throws KeenQueryClientException
      */
     URL getAnalysisUrl(String projectId, String analysisPath) throws KeenQueryClientException {
         try {
             return new URL(String.format(Locale.US,
-                                         "%s/%s/projects/%s/queries/%s",
-                                         this.baseUrl,
-                                         this.apiVersion,
-                                         projectId,
-                                         analysisPath
+                    "%s/%s/projects/%s/queries/%s",
+                    this.baseUrl,
+                    this.apiVersion,
+                    projectId,
+                    analysisPath
             ));
         } catch (MalformedURLException ex) {
             Logger.getLogger(RequestUrlBuilder.class.getName())
-                  .log(Level.SEVERE, "Failed to format query URL.", ex);
+                    .log(Level.SEVERE, "Failed to format query URL.", ex);
 
             throw new KeenQueryClientException("Failed to format query URL.", ex);
         }
     }
+
+    URL getDatasetsUrl(String projectId, String datasetName, boolean fetchResults, Map<String, ?> queryParams) throws KeenQueryClientException {
```

Comment:
Split this method into several methods. It's hard to understand what's going on.